modules/calc/src/main/java/com/opengamma/strata/calc/CalculationRules.java

```diff
@@ -22,88 +23,118 @@
 import com.opengamma.strata.calc.runner.function.CalculationFunction;
 
 /**
  * A set of rules that define how the calculation engine should perform calculations.
  * <p>
- * The calculation rules include:
+ * {@link CalculationRunner} provides the ability to perform calculations on many targets,
+ * such as trades and positions. It returns a grid of results, with the targets as rows.
+ * Each individual calculation is controlled by three things:
  * <ul>
- *   <li>The pricing rules which define how the calculations should be performed. This includes the model
- *   and any model parameters</li>
- *   <li>The market data rules which specify which items of market data should be used for a calculation,
- *   for example which curve group should be used as the source of curves and which market data feed should
- *   be used to request observable values</li>
- *   <li>The reporting currency defines the currency of the result</li>
+ *   <li>The {@linkplain CalculationFunction function}, selected by the target type</li>
+ *   <li>The {@linkplain Measure measure}, the high-level output to be calculated</li>
+ *   <li>The {@linkplain CalculationParameters parameters}, adjust how the measure is to be calculated</li>
  * </ul>
+ * {@code CalculationRules} operates in association with {@link Column}.
+ * The column is used to define the measure. It can also be used to specify column-specific parameters.
+ * The rules contain the complete set of functions and the default set of parameters.
  */
-@BeanDefinition
+@BeanDefinition(builderScope = "private")
 public final class CalculationRules implements ImmutableBean {
 
   /**
-   * The rules defining how calculations should be performed.
+   * The calculation functions.
    * <p>
-   * This controls which {@link CalculationFunction} is used for a calculation.
-   * This defaults to an empty instance if using the builder.
+   * Functions provide the logic of the calculation.
+   * Each type of target must have an associated function in order for calculations to be performed.
    */
-  @PropertyDefinition(validate = "notNull")
-  private final PricingRules pricingRules;
+  @PropertyDefinition
+  private final CalculationFunctions functions;
   /**
-   * The rules that define what market data should be used in each calculation.
+   * The calculation parameters, used to control the how the calculation is performed.
    * <p>
-   * This controls which subset of the complete set of market data should be used for a calculation.
-   * This defaults to an empty instance if using the builder.
-   */
-  @PropertyDefinition(validate = "notNull")
-  private final MarketDataRules marketDataRules;
-  /**
-   * The reporting currency.
+   * Parameters are used to parameterize the {@link Measure} to be calculated.
+   * Each {@link Column} specifies the measure and column-specific parameters.
+   * These parameters are are the defaults that apply to all columns.
```

Comment:
When I read "These parameters are the defaults..." I thought it was referring to the column-specific parameters from the previous sentence and got confused. Maybe it would be clearer to say "The parameters in the calculation rules are the defaults that..."